### PR TITLE
feat: Float.abs function

### PR DESCRIPTION
**Pull request by fynnsu**
This is an implementation of a Float.abs function as mentioned in #1485 .

### CODE REVIEW

File: tests/lean/floatAbs.lean [hunk 1]
```diff
@@ -0,0 +1,7 @@
+def main : IO Unit := do
+IO.println ((3.14159 : Float).abs);
+IO.println ((-3.14159 : Float).abs);
+IO.println ((-2 : Float).abs);
+IO.println ((0 : Float).abs);
```
Comment (digama0) [inline tests/lean/floatAbs.lean:5]: This should probably go in the compiler tests, so it is tested as both compiled and interpreted. I recommend adding it to `float.lean` next to the other float ops. Also, float ops should always be tested on infinities, NaN, and signed zero.

**Comment by leodemoura** [comment]
Thanks for the PR. 
Could you please remove the `stage0` files from the PR? They are not needed.

File: tests/lean/floatAbs.lean (hunk 1 above)
Comment (FynnSu) [reply tests/lean/floatAbs.lean]: I moved the tests as requested. 

I wasn't sure if there is a way to explicitly input -inf or NaN as a float, so I used 1 / 0, and 0 / 0 respectively.
